Userland/Libraries/LibJS/AST.cpp

```diff
@@ -4753,4 +4755,43 @@ ModuleRequest::ModuleRequest(FlyString module_specifier_, Vector<Assertion> asse
     });
 }
 
+// https://html.spec.whatwg.org/multipage/webappapis.html#module-type-from-module-request
+String ModuleRequest::module_type() const
+{
+    // 1. Let moduleType be "javascript".
+    String module_type = "javascript"sv;
+
+    // 2. If moduleRequest.[[Assertions]] has a Record entry such that entry.[[Key]] is "type", then:
+    for (auto const& entry : assertions) {
+        if (entry.key != "type"sv)
+            continue;
+
+        // 1. If entry.[[Value]] is "javascript", then set moduleType to null.
+        if (entry.value == "javascript"sv)
+            module_type = nullptr;
+        // 2. Otherwise, set moduleType to entry.[[Value]].
+        else
+            module_type = entry.value;
+    }
+
+    return module_type;
+}
+
+// https://html.spec.whatwg.org/multipage/webappapis.html#resolve-a-module-specifier
+AK::URL ModuleRequest::resolve_specifier(AK::URL const& base_url) const
```

Comment:
Both of these functions although useful are from the HTML spec and should probably not live here, just as free standing in some part of module fetching in LibWeb seems fine.

Reply:
Not _probably_, _definitely_ :)